desc/geometry/curve.py

```diff
@@ -428,75 +369,58 @@ def Z_n(self, new):
             )
 
     @classmethod
-    def from_values(cls, coords, N=10, grid=None, basis="xyz", name=""):
+    def from_values(cls, coords, N=10, phis=None, basis="xyz", name=""):
         """Fit coordinates to FourierXYZCurve representation.
 
         Parameters
         ----------
+        coords: ndarray
+            coordinates to fit a FourierXYZCurve object with.
         N : int
             Fourier resolution of the new X,Y,Z representation.
             default is 10
-
+        phis : ndarray
+            angle to use for the fitting. if None, defaults to
+            normalized arclength
+        basis : {"rpz", "xyz"}
+            basis for input coordinates. Defaults to "xyz"
         Returns
         -------
         curve : FourierXYZCurve
             New representation of the curve parameterized by Fourier series for X,Y,Z.
 
         """
         if basis == "xyz":
-            coords_rpz = xyz2rpz(coords)
             coords_xyz = coords
         else:
-            coords_rpz = coords
             coords_xyz = rpz2xyz(coords, phi=coords[:, 1])
-        grid = LinearGrid(zeta=coords_rpz[:, 1], NFP=1, sym=False)
+        X = coords_xyz[:, 0]
+        Y = coords_xyz[:, 1]
+        Z = coords_xyz[:, 2]
+
+        assert np.allclose(X[-1], X[0], atol=1e-14), "Must pass in a closed curve!"
```

Comment:
does it need to be closed explicitly? Wouldn't the fit take care of that?

Reply:
I'd rather keep this requirement, reason being if an angle is passed in, it is easier to rescale to [0,2pi] if I know that the last point is the endpoint, otherwise I have no way of knowing of knowing where to put the last point before doing the rescale

(use case: you pass in an arbitrary angle from 0->4 but it is not equally spaced, then I have no way of extrapolating to the endpoint)